### ISSUE TITLE
parallel GPUs working like serial

### ISSUE THREAD

**Issue by ybNo1** [noahchalifour/rnnt-speech-recognition#22]
I find a situation confusing when made 2 time stamp in the code.
when I modify the run_common_voice.py 137~163 like this:

        def step_fn(inputs):
            (mel_specs, pred_inp, 
             spec_lengths, label_lengths, labels) = inputs
            with tf.GradientTape() as tape:
                outputs = model([mel_specs, pred_inp], 
                    training=True)
                # adding line below
                begin_time = time.tme()
                loss = loss_fn(labels, outputs,
                    spec_lengths, label_lengths)
                loss *= (1. / batch_size)

                if fp16_run:
                    loss = optimizer.get_scaled_loss(loss)

            if train_metrics is not None:
                metric_results = run_metrics(mel_specs, labels,
                    metrics=train_metrics)
                metric_results = {name: result * (1. / max(len(gpus), 1)) for name, result in metric_results.items()}

            gradients = tape.gradient(loss, model.trainable_variables)
            if fp16_run:
                gradients = optimizer.get_unscaled_gradients(gradients)
            # adding line below before update gradients
            end_time = time.time()
            print(begin_time, end_time, end_time-begin_time)
            optimizer.apply_gradients(zip(gradients, model.trainable_variables))

            return loss, metric_results

I found that if using 2 GPUs(say gpu0 and gpu1), 
the two gpu printing time_stamp: bg1, ed1 and bg2, ed2 
    **the four params relationship is like this:
    bg1<ed1<bg2<ed2**
that is only when gpu0 finished inference, gpu1 start inferencing 
Was the MirroredStrategy not working?
I commented the tf.function() ,was there any relationship?

**Comment by ybNo1** [comment]
It found to be the right behavior of the eager mode:https://github.com/tensorflow/tensorflow/issues/38924
